### DISCUSSION TITLE
EntityRuler - non-capturing tokens?

### DISCUSSION THREAD

**Discussion by michelleful** [explosion/spaCy#8959] [Help: Coding & Implementations] [answered]
I'm wondering if there is a way to specify that tokens in an EntityRuler should not be captured, e.g.

if I want "problem" but not "no problem", I could specify this pattern:  `[{"LEMMA": "no", "OP": "!"}, {"LEMMA": "problem"}]`.

However, the previous token, so long as it is not "no", would get captured, and furthermore if there is an overlap with another entity, "problem" would not get captured at all:

- "help problem" -> "help problem" is the entity, not "problem"
- "account problem" -> if "account" is a separate entity, then "problem" is not recognised as an entity, due to the non-overlapping entities rule.

This could also come in useful even in non-OP=! situations, e.g. if you wanted to only capture a token in proximity to some other token, but without capturing the second token as part of your entity.

Is there any way to do this currently? If not, what is the recommended way to capture "problem" but not "no problem"? I would prefer to stick to EntityRuler if that's at all possible, but if I have to back off to using Matcher and a callback function, that would be good to know too. Thank you!

ETA: I found a possible solution, which is to define a "no problem" rule with an empty entity label. This seems to work at present but seems like a potentially brittle solution.

**Comment by polm** [comment, marked as answer]
This has been requested before (#6420), but has not been implemented and is not something we are actively working on.

The best way to do this at the moment is to match broadly and post-process results to discard things you don't want. This is not ideal but it's not very hard either. Your placeholder entity solution is also quite clever and sounds reasonable.
